From: Neil Horman <nhorman@redhat.com>
Date: Thu, 14 Jan 2010 20:52:17 -0500
Subject: [net] dccp: fix module load dependency btw dccp_probe and dccp
Message-id: <20100114205217.GJ32577@hmsreliant.think-freely.org>
Patchwork-id: 22551
O-Subject: [RHEL6 PATCH] dccp: fix module load dependency btw dccp_probe and
	dccp (bz 554840)
Bugzilla: 554840
RH-Acked-by: David S. Miller <davem@redhat.com>

Hey all-
        dccp_probe has a silent dependency on dccp.  The former registers a
jprobe on a symbol in the latter module.  Since jprobe_register only referrs to
the symbol as a text string, rather than the address of the symbol itself,
depmod can't see the dependency, and trying to load dccp_probe.ko before dccp.ko
is loaded results in a failed load with -EINVAL returned.  The fix is pretty
easy, just wrap the registration in a call to try_then_request_module, to force
the dependency to get resolved.  I've test/verified this fix, and sent the patch
upstream for review. Acme's acked it, so I expect it will go in. Resolves bz
554840

Neil

Signed-off-by: Aristeu Rozanski <arozansk@redhat.com>

diff --git a/net/dccp/probe.c b/net/dccp/probe.c
index 37731da..4875998 100644
--- a/net/dccp/probe.c
+++ b/net/dccp/probe.c
@@ -164,7 +164,8 @@ static __init int dccpprobe_init(void)
 	if (!proc_net_fops_create(&init_net, procname, S_IRUSR, &dccpprobe_fops))
 		goto err0;
 
-	ret = register_jprobe(&dccp_send_probe);
+	ret = try_then_request_module((register_jprobe(&dccp_send_probe) == 0),
+					"dccp");
 	if (ret)
 		goto err1;
 
